### ISSUE TITLE
Lane Connector isn't working on Underground Metro Tracks

### ISSUE THREAD

**Issue by Seanskyyy** [CitiesSkylinesMods/TMPE#1330]
### Describe the problem
For some off reason, I am unable to utilize the lane connector for underground metro tracks as I had done before. Strangely enough, all prior connections that I've implemented are still active, but are invisible. Typically you'd hover over a white circle and thus edit lane connections. For my underground metro, the circles are orange and when I hover over it this symbol "🚫" pops up.

### Steps to reproduce

1. Load game
2. Click Lane Connector
3. Hover over underground metro tracks

### Log files

N/A

### Savegame?

https://steamcommunity.com/sharedfiles/filedetails/?id=2733863566

### Screenshots?

https://imgur.com/a/Tk4UnnB

### Notes or questions?

N/A

**Comment by aubergine10** [comment]
Were you in underground camera mode when trying to set lane connectors? (PageDown = undeground view)

**Comment by krzychu124** [comment]
@Seanskyyy It's because of you opened transport panel. Underground mode when transport panel is open is slightly different than this used by TM:PE when switched with PageDown. I'll add that one too. Thanks for heads up.

**Comment by krzychu124** [comment]
@aubergine10 should we disable the active tool when Info mode was changed?

**Comment by aubergine10** [comment]
Not sure. I'll do some testing in-game to see how it "feels" if the tool were de-activated on info view change.

Regarding making it work in different info views, is that primarily a case of "allow both overground and underground when an infoview is open" or something more complex?

**Comment by krzychu124** [comment]
> "allow both overground and underground when an infoview is open"

I hope it will be that simple, haven't checked the code yet but there are many info modes with underground

**Comment by aubergine10** [comment]
Testing in-game, opening TM:PE toolbar, when opened via main button, closes current infoview. However, if TM:PE toolbar is already open, then opening an infoview won't close the TM:PE toolbar.

What I'd expect:

* Opening TM:PE toolbar should close most info views (currently it seems to close any/all infoviews)
* Opening most infoviews should close TM:PE toolbar (and any active tool that was in use).
* Possible exceptions are:
    * Traffic
    * Transport? Would be useful to alter vehicle restrictions while working with bus routes in particular
    * Traffic Routes - TM:PE [Adjust Roads Panel](https://github.com/CitiesSkylinesMods/TMPE/wiki/Adjust-Roads) is operational in Traffic Routes view (regardless of whether TM:PE toolbar is visible)

tl;dr: TM:PE + info views = generally NOPE, except for the possible exceptions above

**Comment by aubergine10** [comment]
Another info view where TM:PE might be useful is Escape Routes (Natural Disasters DLC)

And possibly Tours (Park Life DLC)?

**Comment by krzychu124** [comment]
Cool, I'll create later a fix/improvement if you don't do it first 😄  Going to sleep 🛏️ 💤

**Comment by aubergine10** [comment]
same lol

**Comment by aubergine10** [comment]
`TrafficManagerTool.cs`:

* `ThreadingExtension.ModUi.CloseMainMenu()`
* `OnDisable()` reverts info manager (only if TMPE was in underground mode though):
    * InfoManager.instance.SetCurrentMode(InfoManager.InfoMode.None, InfoManager.SubInfoMode.Default);

Info manager:

`private InfoMode m_currentMode;`
`InfoManager.instance.CurrentMode`

`InfoModeChangedEventHandler(InfoMode mode, SubInfoMode subMode);`

`public event InfoModeChangedEventHandler EventInfoModeChanged;`

`InfoMode.`:

* `Transport` = public transport - already cluttered view, close toolbar?
    * Future: Could possibly show panel with bus-affecting options?
* `Connections` = outside connections - uncluttered, allow toolbar?
* `Traffic` = traffic congestion - allow toolbar
* `Maintenance` = road maintenance - allow toolbar?
    * Future: Show affect on speed limits?
* `Snow` = snow coverage - allow toolbar?
    * Future: Show affect on speed limits?
* `EscapeRoutes` = emergency shelter routes
    * Future: Show evac bus 'ingore rules' state?
* `TrafficRoutes` = info view where bulk applicators are available - allow toolbar

**Comment by aubergine10** [comment]
working on it...

**Comment by aubergine10** [comment]
PR ready for review: #1333
